test/rules/no-irregular-whitespaces/test.ts.lint

```diff
@@ -0,0 +1,50 @@
+letfoo;
+   ~     [Irregular whitespace not allowed]
+letfoo;
+   ~     [Irregular whitespace not allowed]
+letfoo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let﻿foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let᠎foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let​foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let　foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
+let foo;
+   ~     [Irregular whitespace not allowed]
```

Comment:
also add a test case with multiple consecutive whitespaces
and a test with regular whitespace